packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -62,7 +64,7 @@ const createNetwork = () => {
       };
       openConnections.add(openFetchConnection);
 
-      res = await fetchPromise;
+      res = await proxyifyPromise(fetchPromise);
```

Comment:
We also need to wrap the `res.body`, more specifically `res.text()`, `res.json()`, `res.arrayBuffer()`, `res.blob()`, `res.formData()`, and make it also work whenever `res.clone()` is used.

And probably the hardest is to wrap the ReadableStream itself. `res.body.pipeThrough()`, `res.body.pipeTo()`, `res.body.tee()` and `res.body.getReader().read()`  should all stop sending data after teardown without closing. [`new Proxy()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) might help, or you can look at the [WebSocket wrapper](https://github.com/MetaMask/snaps-skunkworks/blob/24dcdfffcc18ab76c0f5c12fb02b8407cfe2226f/packages/execution-environments/src/common/endowments/network.ts#L104)

We don't really have [tests for the body](https://github.com/MetaMask/snaps-skunkworks/blob/24dcdfffcc18ab76c0f5c12fb02b8407cfe2226f/packages/execution-environments/src/common/endowments/network.test.ts#L61), but if you're feeling ambitious, you could write some e2e tests there. But that's not necessary for this PR.

Reply:
I've wrapped all the promises. Only streams are left which are difficult to test at the moment, because of the setup with jest. I'm creating a new ticket only for this problem as discussed on the last standup.

Ticket is here: https://github.com/MetaMask/snaps-skunkworks/issues/681

This way we can unblock and finish everything else that's usable in this PR and possibly merge it soon when ready.